maven/lib/dependabot/maven/file_fetcher.rb

```diff
@@ -122,6 +133,28 @@ def parent_path_for_pom(pom)
 
         Pathname.new(File.join(*name_parts)).cleanpath.to_path
       end
+
+      def fetched_pom_is_parent(pom, parent_pom)
+        pom_doc = Nokogiri::XML(pom.content).remove_namespaces!
```

Comment:
Oh weird, `remove_namespaces!` actually returns something? I'm used to the exclamation point variant does an in-place transform and there's a similarly named method with no exclamation point that must be called when wanting to return something... Otherwise`pom_doc` ends up `nil`.

I'm on my phone so can't look up the docs, but wanted to comment so at least one of us remembers to double check... Although I assume this is covered by a test in at least some manner.

But I'm also not a ruby expert so may be misunderstanding something...

Reply:
That is the same reaction i had (but i'm surely not a ruby expert as well) 😃 And apparently there is no `remove_namespaces` method there as well. But then i have checked the sources for initila method here https://www.rubydoc.info/github/sparklemotion/nokogiri/Nokogiri%2FXML%2FDocument:remove_namespaces! and it indeed looks like it returns self in the end. And the tests would have indeed failed if it did not return the XML doc back.

Reply:
This suprised me too!